### PR TITLE
Update LSRA Doc

### PR DESCRIPTION
**Pull request by CarolEidt**

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

**Comment by kunalspathak** [comment]
While you are there, do you mind updating the table of contents as well? Here is what I extracted from your file:

```

  * [Overview](#overview)
  * [Preconditions](#preconditions)
    + [Lowered IR Form (LIR)](#lowered-ir-form-lir)
    + [Register Requirements](#register-requirements)
  * [Post-Conditions](#post-conditions)
  * [LSRA Phases](#lsra-phases)
    + [Liveness and Candidate Identification](#liveness-and-candidate-identification)
    + [Block Ordering](#block-ordering)
    + [Building Intervals and RefPositions](#building-intervals-and-refpositions)
    + [Register allocation (doLinearScan)](#register-allocation-dolinearscan)
  * [Key Data Structures](#key-data-structures)
    + [Live In](#live-in)
    + [currentLiveVars](#currentlivevars)
    + [Referenceable](#referenceable)
    + [Interval](#interval)
    + [RegRecord](#regrecord)
    + [RefPosition](#refposition)
    + [GenTree Nodes](#gentree-nodes)
    + [VarToRegMap](#vartoregmap)
  * [Dumps and Debugging Support](#dumps-and-debugging-support)
  * [LSRA Stress Modes](#lsra-stress-modes)
  * [Assertions & Validation](#assertions--validation)
  * [Future Extensions and Enhancements](#future-extensions-and-enhancements)
  * [Feature Enhancements](#feature-enhancements)
    + [Support for Allocating Consecutive Registers](#support-for-allocating-consecutive-registers)
  * [Code Quality Enhancements](#code-quality-enhancements)
    + [Merge Allocation of Free and Busy Registers](#merge-allocation-of-free-and-busy-registers)
    + [Auto-tuning of register selection](#auto-tuning-of-register-selection)
    + [Pre-allocating high frequency lclVars](#pre-allocating-high-frequency-lclvars)
    + [Avoid Splitting Loop Backedges](#avoid-splitting-loop-backedges)
    + [Enable EHWriteThru by default](#enable-ehwritethru-by-default)
    + [Avoid Spill When Stack Copy is Valid](#avoid-spill-when-stack-copy-is-valid)
    + [Rematerialization](#rematerialization)
    + [Improving Reg-Optional Support](#improving-reg-optional-support)
      - [Reg-Optional Defs](#reg-optional-defs)
      - [Don't Pre-determine Reg-Optional Operand](#dont-pre-determine-reg-optional-operand)
      - [Don't Mark DelayFree for Duplicate Operands](#dont-mark-delayfree-for-duplicate-operands)
    + [Improving Preferencing](#improving-preferencing)
    + [Leveraging SSA form](#leveraging-ssa-form)
    + [Spanning trees for physical registers](#spanning-trees-for-physical-registers)
    + [Improve the handling of def/use conflicts](#improve-the-handling-of-defuse-conflicts)
  * [Throughput Enhancements](#throughput-enhancements)
    + [Allocation Window for Min-Opts and Tier 0](#allocation-window-for-min-opts-and-tier-0)
    + [Distinguish Intra-Block versus Inter-Block Variables](#distinguish-intra-block-versus-inter-block-variables)
    + [Improve the VarToRegMap](#improve-the-vartoregmap)
    + [Other Throughput Investigations](#other-throughput-investigations)
  * [Test and Cleanup Issues](#test-and-cleanup-issues)
  * [References](#references)
```

File: docs/design/coreclr/jit/lsra-detail.md [hunk 1]
```diff
@@ -567,16 +589,15 @@ LinearScanAllocation(List<RefPosition> refPositions)
         -   Whether the `RefPosition` being allocated, or the one
             potentially being spilled, is reg-optional
 
+    -   Both `tryAllocateFreeReg()` and `allocateBusyReg()` currently fully evaluate the "goodness"
+        of each  register.
+
     -   It will always spill an `Interval` either at its most recent
         use, or at the entry to the current block.
 
-        -   Issues [\#7609](https://github.com/dotnet/coreclr/issues/7609) and
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:573]: Any reason for deleting the pointer to the tracking issues?

File: docs/design/coreclr/jit/lsra-detail.md [hunk 2]
```diff
@@ -895,18 +905,15 @@ exclusive:
 
 -   Never allocate a register for a `RefPosition` marked `regOptional` (0x1000).
 
-It may be useful to also have a stress mode that deliberately trashes registers that
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:898]: same here.

File: docs/design/coreclr/jit/lsra-detail.md (hunk 1 above)
Comment (kunalspathak) [reply docs/design/coreclr/jit/lsra-detail.md:573]: Ah, I see that you have moved them down. Never mind.

---
In reply to: [457638736](https://github.com/dotnet/runtime/pull/39492#discussion_r457638736) [](ancestors = 457638736)

**Review by kunalspathak — APPROVED** [review]
LGTM

**Review by sandreenko — APPROVED** [review]
LGTM, thanks for keeping it up-to-date.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**